test/functional/job/job_spec.lua

```diff
@@ -37,6 +37,31 @@ describe('jobs', function()
     eq({'notification', 'exit', {0, 0}}, next_msg())
   end)
 
+  it('changes to / when cwd provided', function()
+    nvim('command', "let g:job_opts.cwd = '/'")
+    nvim('command', "let j = jobstart('pwd', g:job_opts)")
+    eq({'notification', 'stdout', {0, {'/', ''}}}, next_msg())
+    eq({'notification', 'exit', {0, 0}}, next_msg())
+  end)
+
+  it('changes to random directory when cwd provided', function()
+    local dir = eval('tempname()')
```

Comment:
This should fix the OSX issue (while being harmless on other platforms):

    local dir = eval('resolve(tempname())')